src/Lykke.AlgoStore.Api/Controllers/AlgoStoreUserPermissionsController.cs

```diff
@@ -0,0 +1,158 @@
+﻿using Lykke.AlgoStore.Api.Infrastructure.Attributes;
+using Lykke.AlgoStore.Api.Models;
+using Lykke.AlgoStore.Core.Domain.Entities;
+using Lykke.AlgoStore.Core.Services;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Mvc;
+using Swashbuckle.AspNetCore.SwaggerGen;
+using System.Collections.Generic;
+using System.Net;
+using System.Threading.Tasks;
+
+namespace Lykke.AlgoStore.Api.Controllers
+{
+    [Authorize]
+    [Route("api/v1/permissions")]
+    public class AlgoStoreUserPermissionsController: Controller
+    {
+        private readonly IUserRolesService _userRolesService;
+        private readonly IUserPermissionsService _permissionsService;
+
+        public AlgoStoreUserPermissionsController(
+            IUserRolesService userRolesService,
+            IUserPermissionsService permissionsService)
+        {
+            _userRolesService = userRolesService;
+            _permissionsService = permissionsService;
+        }
+
+        [HttpGet("getAll")]
+        [RequirePermission]
+        [SwaggerOperation("GetAllpermissions")]
+        [ProducesResponseType(typeof(List<UserPermissionModel>), (int)HttpStatusCode.OK)]
+        [ProducesResponseType(typeof(BaseErrorResponse), (int)HttpStatusCode.InternalServerError)]
+        [ProducesResponseType((int)HttpStatusCode.NotFound)]
+        public async Task<IActionResult> GetAllpermissions()
+        {
+            var result = await _permissionsService.GetAllPermissionsAsync();
+
+            return Ok(result);
+        }
+
+        [HttpGet("getById")]
+        [RequirePermission]
+        [SwaggerOperation("GetPermissionById")]
+        [ProducesResponseType(typeof(UserPermissionModel), (int)HttpStatusCode.OK)]
+        [ProducesResponseType(typeof(BaseErrorResponse), (int)HttpStatusCode.InternalServerError)]
+        [ProducesResponseType((int)HttpStatusCode.NotFound)]
+        public async Task<IActionResult> GetPermissionById(string permissionId)
+        {
+            var result = await _permissionsService.GetPermissionByIdAsync(permissionId);
```

Comment:
in this endpoint we return only "Ok" Result, so 
[ProducesResponseType(typeof(BaseErrorResponse), (int)HttpStatusCode.InternalServerError)]
is not requiered, or we need to return BadResponse too.